astropy/coordinates/baseframe.py

```diff
@@ -473,7 +555,7 @@ def __init__(self, *args, copy=True, representation_type=None,
                                      'velocity differential is supported. Got: '
                                      '{}'.format(diffs))
 
-        elif self.representation_type:
+        else:
```

Comment:
With the changes in `__init__`, `self.representation_type` is always set, or should be